ui/e2e/src/fixtures/dashboardTest.ts

```diff
@@ -20,23 +21,63 @@ type DashboardTestOptions = {
 
 type DashboardTestFixtures = {
   dashboardName: string;
+
+  /**
+   * Set to true if the test set will modify the dashboard. When true, the
+   * dashboard will be reset at the end of each test. Do not use in read-only
+   * tests because it will unnecessarily slow them down.
+   */
+  modifiesDashboard: boolean;
   dashboardPage: DashboardPage;
 };
 
+const BACKEND_BASE_URL = 'http://localhost:8080';
+
+async function getDashboardJson(projectName: string, dashboardName: string) {
+  const queryUrl = `${BACKEND_BASE_URL}/api/v1/projects/${projectName}/dashboards/${dashboardName}`;
+  const results = await fetch(queryUrl);
+  const dashboardJson = await results.json();
+  return dashboardJson;
+}
+
+async function setDashboardJson(projectName: string, dashboardName: string, content: unknown) {
+  const queryUrl = `${BACKEND_BASE_URL}/api/v1/projects/${projectName}/dashboards/${dashboardName}`;
+  const result = await fetch(queryUrl, {
+    method: 'PUT',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(content),
+  });
+  return result;
```

Comment:
I guess we don't need to `result.json()` this result?

Reply:
We shouldn't need it in a few of these cases. We really only need the result body when we're fetching the dashboard content.